Source/WTF/wtf/CheckedArithmetic.h

```diff
@@ -1013,6 +1013,12 @@ template<typename T, typename U> bool differenceOverflows(U left, U right)
     return checkedDifference<T>(left, right).hasOverflowed();
 }
 
+template<typename T> T sumIfNoOverflow(T left, T right)
```

Comment:
As Darin mentioned:
`Interestingly, checkedSum is almost what we want, in fact it even stores the first value if adding the second value overflows. But you can’t get that value out without the code intentionally crashing in the Checked::value function!`

So, I could use `checkedSum()` locally and explicitly checking for overflow. I've done it similarly before using a lambda, but Darin asked it to be moved to a header. Maybe we need a better name though:

`I think that having sumOrFirstValue as a local lambda here is not great since this seems like the kind of operation we’d want to use more than once. I’d want it to be in CheckedArithmetic.h`